### PR TITLE
Introduce update state on CV implementation API

### PR DESCRIPTION
**Pull request by mandelsoft**
#### What this PR does / why we need it

Extend the CV implementation API to provide information, whether an Update of the descriptor has been done.
This information is then used to trigger the pubsub handler.
If no update has been done, the query of the repository's pubsub info is omitted.

### CODE REVIEW

File: api/ocm/extensions/repositories/composition/repository.go [hunk 1]
```diff
@@ -173,25 +173,28 @@ func (v *VersionAccess) AddBlob(blob cpi.BlobAccess) (string, error) {
...
 }
 
-func (v *VersionAccess) Update() error {
+func (v *VersionAccess) Update() (bool, error) {
 	v.access.lock.Lock()
 	defer v.access.lock.Unlock()
 
 	if v.readonly {
-		return accessio.ErrReadOnly
+		return true, accessio.ErrReadOnly
```
Comment (fabianburth) [inline api/ocm/extensions/repositories/composition/repository.go:181]: Why do we return `true` here?

File: api/ocm/extensions/repositories/virtual/example/example.go [hunk 2]
```diff
@@ -192,30 +192,35 @@ func (v *VersionAccess) AddBlob(blob cpi.BlobAccess) (string, error) {
...
 
 	if v.desc.GetName() != v.comp || v.desc.GetVersion() != v.vers {
-		return errors.ErrInvalid(cpi.KIND_COMPONENTVERSION, common.VersionedElementKey(v.desc).String())
+		return false, errors.ErrInvalid(cpi.KIND_COMPONENTVERSION, common.VersionedElementKey(v.desc).String())
 	}
 	i := v.access.index.Get(v.comp, v.vers)
 	if !reflect.DeepEqual(v.desc, i.CD()) {
 		if v.IsReadOnly() {
-			return accessio.ErrReadOnly
+			return true, accessio.ErrReadOnly
```
Comment (fabianburth) [inline api/ocm/extensions/repositories/virtual/example/example.go:205]: same thing here?

**Review by fabianburth — CHANGES_REQUESTED** [review]

**Comment by fabianburth** [comment]
are there tests for this behaviour?

File: api/ocm/extensions/repositories/composition/repository.go (hunk 1 above)
Comment (mandelsoft) [reply api/ocm/extensions/repositories/composition/repository.go:181]: fixed for all such returns

**Comment by frewilhelm** [comment]
Will this fix https://github.com/open-component-model/ocm/issues/872?

**Comment by jakobmoellerdev** [comment]
just needs a rebase and a rereview

**Review by jakobmoellerdev — APPROVED** [review]
LGTM

**Comment by jakobmoellerdev** [comment]
> Will this fix #872?

no @frewilhelm

**Review by hilmarf — APPROVED** [review]

**Pull request merged**